jerry-core/vm/vm.c

```diff
@@ -794,92 +769,70 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
             break;
           }
         }
-        if (CBC_BRANCH_IS_BACKWARD (opcode_flags))
+
+        if (opcode_data & VM_OC_BACKWARD_BRANCH)
         {
           branch_offset = -branch_offset;
         }
       }
 
-      free_flags = 0;
+      left_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+      right_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+
       if (VM_OC_HAS_GET_ARGS (opcode_data))
       {
-        uint32_t operands = VM_OC_GET_ARGS_GET_INDEX (opcode_data);
+        uint32_t operands = VM_OC_GET_ARGS_INDEX (opcode_data);
 
-        if (operands >= VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_LITERAL))
+        if (operands >= VM_OC_GET_LITERAL)
         {
           uint16_t literal_index;
           READ_LITERAL_INDEX (literal_index);
           READ_LITERAL (literal_index,
-                        left_value,
-                        free_flags = VM_FREE_LEFT_VALUE);
+                        left_value);
 
           switch (operands)
           {
-            case VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_STACK_LITERAL):
-            {
-              JERRY_ASSERT (stack_top_p > frame_ctx_p->registers_p + register_end);
-              right_value = left_value;
-              left_value = *(--stack_top_p);
-              free_flags = (uint8_t) ((free_flags << 1) | VM_FREE_LEFT_VALUE);
-              break;
-            }
-            case VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_LITERAL_BYTE):
+            case VM_OC_GET_LITERAL:
```

Comment:
I'm not sure you can control switch layout at this level. I just built this PR and on x86_64 in release mode, and got the following:
```asm
  402b17:       66 81 7c 24 50 00 02    cmpw   $0x200,0x50(%rsp) ; VM_OC_GET_LITERAL_LITERAL
  402b1e:       74 35                   je     402b55 <vm_loop+0x2c1>
  402b20:       66 81 7c 24 50 80 02    cmpw   $0x280,0x50(%rsp) ; VM_OC_GET_STACK_LITERAL
  402b27:       0f 84 35 01 00 00       je     402c62 <vm_loop+0x3ce>
  402b2d:       66 81 7c 24 50 80 01    cmpw   $0x180,0x50(%rsp) ; VM_OC_GET_LITERAL
  402b34:       c7 04 24 18 00 00 00    movl   $0x18,(%rsp)
  402b3b:       0f 84 8a 01 00 00       je     402ccb <vm_loop+0x437>
```
The compiler generated code to check for `VM_OC_GET_LITERAL` as the third step. Is `VM_OC_GET_LITERAL` really that frequent to make it worth factor out into an enclosing `if`?